state/application.go

```diff
@@ -284,6 +284,17 @@ func (op *DestroyApplicationOperation) Build(attempt int) ([]txn.Op, error) {
 
 // Done is part of the ModelOperation interface.
 func (op *DestroyApplicationOperation) Done(err error) error {
+	if err == txn.ErrAborted {
+		connected, err2 := applicationHasConnectedOffers(op.app.st, op.app.Name())
+		if err2 != nil {
+			err = errors.Trace(err2)
+		} else if connected {
+			err = errors.NotSupportedf("newly related offer")
```

Comment:
These error messages will ultimately bubble up to the user - they're not very helpful.
You need to catch these at the CLI and display something better.

Also, the "newly related offer" message above is simply the issue that there are offers on he application so we'd just want to return "application is used by %d offer%s", n, plural(n)"